src/compiler/program.ts

```diff
@@ -903,11 +903,12 @@ namespace ts {
                     return;
                 }
 
+                const caseSensitive = host.useCaseSensitiveFileNames();
                 for (let i = 0, n = Math.min(commonPathComponents.length, sourcePathComponents.length); i < n; i++) {
-                    if (commonPathComponents[i] !== sourcePathComponents[i]) {
+                    if (caseSensitive ? commonPathComponents[i] !== sourcePathComponents[i] : commonPathComponents[i].toLocaleLowerCase() !== sourcePathComponents[i].toLocaleLowerCase()) {

```

Comment:
Rolled back using `getCanonicalFileName` - the change in output this caused was a nightmare for our projects tests. On first attempt, the canonical path becomes absolute in many cases - this was a big change which would pretty much require reworking the test harness' path handling, so then I tried patching the function to make the result relative if need be - this was _still_ a change in output, because the resulting _relative paths_ could be different than the ones found by looking for common path components (usually they'd end up with more common components). This caused absolute paths to be generated by other algorithms elsewhere which then got emitted in tests and, generally, it was a mess.

We should _really_ swap to using fully qualified `Path` types for everything and convert to root-relative paths for test output (in the harness).